source/slang/slang-ir-redundancy-removal.cpp

```diff
@@ -126,6 +126,197 @@ bool removeRedundancy(IRModule* module, bool hoistLoopInvariantInsts)
     return changed;
 }
 
+bool isAddressMutable(IRInst* inst)
+{
+    auto rootType = getRootAddr(inst)->getDataType();
+    switch (rootType->getOp())
+    {
+    case kIROp_ParameterBlockType:
+    case kIROp_ConstantBufferType:
+    case kIROp_StructuredBufferLoad:
+    case kIROp_GetStructuredBufferPtr:
```

Comment:
We should consider StructuredBuffer as mutable by default, since the resources may alias. There could be another RWStructuredBuffer pointing to the same memory location as the structured buffer.